test/unit_tests/arkode/C_serial/ark_test_adapt.c

```diff
@@ -0,0 +1,154 @@
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------------------
+ * Unit test for step size growth and handling of inf/nan within a controller
+ * due to 0 local error.
+ * ---------------------------------------------------------------------------*/
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "arkode/arkode_erkstep.h"
+#include "nvector/nvector_serial.h"
+
+/* If an error occurs, print to stderr and exit */
+static void err_fn(const int line, const char* const func, const char* const file,
+                   const char* const msg, const SUNErrCode err_code,
+                   void* const err_user_data, const SUNContext sunctx)
+{
+  fprintf(stderr, "Error at line %i of %s in %s: %s\n", line, func, file, msg);
+  exit(err_code);
+}
+
+/* RHS for the simple ODE y'=0 */
+static int f(const sunrealtype t, const N_Vector y, const N_Vector ydot,
+             void* const user_data)
+{
+  N_VConst(SUN_RCONST(0.0), ydot);
+  return 0;
+}
+
+/* Take a single time step solving y'=0 and check the max growth was attained */
+static int check_step(void* const arkode_mem, const N_Vector y,
+                      const sunrealtype h_expected, const int step)
+{
+  /* Integration much farther than expected step */
+  const sunrealtype tout = SUN_RCONST(100.0) * h_expected;
+  sunrealtype tret;
+  /* The ERK method should be able to take the maximum possible timestep for the
+     without any rejected steps or local error */
+  ARKodeEvolve(arkode_mem, tout, y, &tret, ARK_ONE_STEP);
+
+  long int local_err_fails;
+  ARKodeGetNumErrTestFails(arkode_mem, &local_err_fails);
+  if (local_err_fails != 0)
+  {
+    fprintf(stderr, "Expected 0 local error failures at step %i but is %li\n",
+            step, local_err_fails);
+  }
+
+  const N_Vector err = N_VClone(y);
+  ARKodeGetEstLocalErrors(arkode_mem, err);
+  const sunrealtype err_norm = N_VMaxNorm(err);
+  N_VDestroy(err);
+
+  if (err_norm != 0)
+  {
+    fprintf(stderr, "Expected local error at step %i to be 0 but is %g\n", step,
+            err_norm);
+    return 1;
+  }
+
+  sunrealtype h_actual;
+  ARKodeGetCurrentStep(arkode_mem, &h_actual);
+  if (SUNRCompare(h_expected, h_actual))
+  {
+    fprintf(stderr, "Expected h at step %i to be %g but is %g\n", step,
+            h_expected, h_actual);
+    return 1;
+  }
+
+  return 0;
+}
+
+/* Take several steps solving y'=0 and check the max growth was attained */
+static int check_steps(void* const arkode_mem, const N_Vector y,
+                       const sunrealtype h0, const sunrealtype first_growth,
+                       const sunrealtype growth)
+{
+  ARKodeSetInitStep(arkode_mem, h0);
+  ARKodeSetMaxFirstGrowth(arkode_mem, first_growth);
+  ARKodeSetMaxGrowth(arkode_mem, growth);
+
+  sunrealtype h_expect = first_growth * h0;
+  /* Take several steps to see the special behavior at step one then to allow
+     the adaptivity controller history fill up */
```

Comment:
Should this be reworded?  I find it confusing as written.
```suggestion
  /* Take several steps to see the special behavior at step one then to allow
     the adaptivity controller history fill up */
```